src/mui/list/List.js

```diff
@@ -99,11 +101,32 @@ export class List extends Component {
         this.props.changeListParams(this.props.resource, newParams);
     }
 
+    exportClicked = (event) => {
+        event.stopPropagation();
+        const query = this.getQuery();
+
+        this.props.push({
+            ...this.props.location,
+            query: {
+                ...query,
+                filter: JSON.stringify(query.filter),
+                perPage: 100000,
```

Comment:
We should leave the `perPage` empty, and leave the interpretation of that parameter to the `restClient`